README

```diff
@@ -35,14 +58,72 @@ Connect client:
-  streams, connection negotiation and establishment, multi-rail, and handling 
-  different memory types.
-- UCT - UCX Transport. Implements low-level communication primitives such as
-  active messages, remote memory access, and atomic operations.
-- UCS - UCX Services. A collection of data structures, algorithms, and system
-  utilities for common use.
-- UCM - UCX Memory. Intercepts memory allocation and release events, used by
-  memory registration cache.
+## Our Community
+
+* [Project Website](http://www.openucx.org/)
+* [Github](http://www.github.com/openucx/ucx/)
+* [Software Releases](http://www.github.com/openucx/ucx/releases)
+* [Mailing List](https://elist.ornl.gov/mailman/listinfo/ucx-group)
+
+## Licenses
+
+UCX comes with 2 licenses:
+
+* [BSD3](LICENSE)
+* [Apache 2.0](LICENSES.apache20)
+
+## Contributor Agreement and Guidelines
+
+In order to contribute to UCX, please sign up with an appropriate
+[Contributor Agreement](http://www.openucx.org/license/).
+
+Follow these
+[instructions](https://github.com/openucx/ucx/blob/master/CONTRIBUTING.md)
+when submitting contributions and changes.
+
+## UCX Publications
+
+To reference UCX in a publication, please use the following entry:
+
+```bibtex
+@inproceedings{shamis2015ucx,
+  title={UCX: an open source framework for HPC network APIs and beyond},
+  author={Shamis, Pavel and Venkata, Manjunath Gorentla and Lopez, M Graham and Baker, Matthew B and Hernandez, Oscar and Itigin, Yossi and Dubman, Mike and Shainer, Gilad and Graham, Richard L and Liss, Liran and others},
+  booktitle={2015 IEEE 23rd Annual Symposium on High-Performance Interconnects},
+  pages={40--43},
+  year={2015},
+  organization={IEEE}
+}
+```
+
+To reference the UCX website:
+
+```bibtex
+@misc{openucx-website,
+    title = {{The Unified Communication X Library}},
+    key = {{{The Unified Communication X Library}},
+    howpublished = {{\url{http://www.openucx.org}}}
+}
+```
+
+## UCX Architecture
+
+<img src="doc/doxygen/Architecture.png">
+
+| Component | Role        | Description |
+| :---:     | :---:       | ---         |
+| UCP | Protocol          | Implements high-level abstractions such as tag-matching, streams, connection negotiation and establishment, multi-rail, and handling different memory types |
+| UCT | Transport         | Implements low-level communication primitives such as active messages, remote memory access, and atomic operations |
+| UCS | Services          | A collection of data structures, algorithms, and system utilities for common use |
+| UCM | Memory            | Intercepts memory allocation and release events, used by the  memory registration cache |
+
+## Supported Transports
+
+* [Infiniband](https://www.infinibandta.org/)
+* [RoCE](http://www.roceinitiative.org/)
+* [Cray Gemini and Aries](https://www.cray.com/)
+* [CUDA](https://developer.nvidia.com/cuda-zone)
+* [ROCm](https://rocm.github.io/)
+* Shared Memory
+    * posix, sysv, cma, knem, and xpmem
+* TCP/IP fallback
```

Comment:
maybe just TCP/IP, without "fallback" word?

Reply:
Our TCP support is first class citizen